crypto-ffi/src/generic.rs

```diff
@@ -719,13 +721,27 @@ pub trait CoreCryptoCallbacks: std::fmt::Debug + Send + Sync {
     ) -> bool;
 }
 
+static INIT_LOGGER: Once = Once::new();
+static LOGGER: LazyLock<ReloadLog<CoreCryptoLoggerWrapper>> = LazyLock::new(|| {
+    ReloadLog::new(CoreCryptoLoggerWrapper {
+        logger: Arc::new(DummyLogger {}),
+    })
+});
+
 /// Initializes the logger
-/// WARNING: This is a global setting. Calling it twice will cause errors.
 #[uniffi::export]
-pub fn set_logger(logger: std::sync::Arc<dyn CoreCryptoLogger>, level: CoreCryptoLogLevel) {
-    let logger = CoreCryptoLoggerWrapper { logger };
-    log::set_boxed_logger(Box::new(logger)).unwrap();
-    log::set_max_level(LevelFilter::from(level));
+pub fn set_logger(logger: Arc<dyn CoreCryptoLogger>) {
+    LOGGER.handle().replace(CoreCryptoLoggerWrapper { logger }).unwrap();
```

Comment:
Shouldn't we properly handle the error? I know its lock poison error, but better safe than sorry

Reply:
Yes, we should 👍

Reply:
How would you suggest we handle it? Start returning an error in this method? That would be a breaking change.

Reply:
Added a commented to explain the unwrapping.